src/device/device-management.ts

```diff
@@ -70,20 +72,18 @@ export async function createCameraTrack<T extends LocalTrack>(
       `Failed to create camera track ${error}`
     );
   }
-  return new constructor(stream);
+  return new LocalCameraStream(stream);
 }
 
 /**
- * Creates a microphone audio track.
+ * Creates a LocalMicrophoneStream with the given constraints.
  *
- * @param constructor - Constructor for the local microphone track.
  * @param constraints - Audio device constraints.
  * @returns A LocalTrack object or an error.
  */
-export async function createMicrophoneTrack<T extends LocalTrack>(
```

Comment:
The `<T extends LocalTrack>` may need to stay in, I think it's used in some cases to create a custom type of stream wrapper.  Maybe `<T extends LocalMicrophoneStream>`?

Reply:
The plan is to do this differently, where extend of extending these tracks (which then means it's disconnected from the rest of the hierarchy)--they'd instead create and wrap this track, so if all goes well then the `T extends XXX` thing will not be needed.

Reply:
Cool, sounds good.